hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -102,14 +94,17 @@ public void initialize(String inputName, Map<String, String> properties) {
       this.conf.set(HiveConf.ConfVars.METASTOREWAREHOUSE.varname, properties.get(CatalogProperties.WAREHOUSE_LOCATION));
     }
 
-    int clientPoolSize = PropertyUtil.propertyAsInt(properties,
-        CatalogProperties.CLIENT_POOL_SIZE, CatalogProperties.CLIENT_POOL_SIZE_DEFAULT);
-    this.clients = new HiveClientPool(clientPoolSize, this.conf);
-    this.createStack = Thread.currentThread().getStackTrace();
-    this.closed = false;
+    this.conf.setInt(CatalogProperties.CLIENT_POOL_SIZE, PropertyUtil.propertyAsInt(properties,
+            CatalogProperties.CLIENT_POOL_SIZE, CatalogProperties.CLIENT_POOL_SIZE_DEFAULT));
+
+    this.conf.setLong(CatalogProperties.CLIENT_POOL_CACHE_EVICTION_INTERVAL_MS,
+            PropertyUtil.propertyAsLong(properties, CatalogProperties.CLIENT_POOL_CACHE_EVICTION_INTERVAL_MS,
```

Comment:
Is there a particular reason to not push this logic to the `HiveClientPoolProvider` and add a property map as an input parameter?
Or otherwise specifically providing it as a constructor parameter?

Reply:
Done.